core-api/src/test/java/com/optimizely/ab/event/internal/UserEventFactoryTest.java

```diff
@@ -60,11 +61,28 @@
 
     private Experiment experiment;
     private Variation variation;
+    private DecisionMetadata decisionMetadata;
 
     @Before
     public void setUp() {
         experiment = new Experiment(EXPERIMENT_ID, EXPERIMENT_KEY, LAYER_ID);
         variation = new Variation(VARIATION_ID, VARIATION_KEY);
+        decisionMetadata = new DecisionMetadata(EXPERIMENT_KEY, EXPERIMENT_KEY, "experiment", VARIATION_KEY);
```

Comment:
Can we change these tests with flagKey? We do not use expKey for flagKey any more.

Reply:
yeah, we will never pass experiment key twice